### PR TITLE
Fixes #7

### LINKED ISSUE #7: False positive when matching on a file extension
I'll add a failing unit test for this in my fork.

### PR DESCRIPTION
**Pull request by dazinator**
Fixes an issue where matching is matching the start of a string, rather than the whole string.
Unit / Regression test passing.

### CODE REVIEW

File: src/Glob/Glob.cs [hunk 1]
```diff
@@ -30,7 +30,7 @@ private void Compile()
             var parser = new Parser(this.Pattern);
             _root = parser.Parse();
 
-            _regex = new Regex(GlobToRegexVisitor.Process(_root));
+            _regex = new Regex(GlobToRegexVisitor.Process(_root) + "$");
```
Comment (kthompson) [inline src/Glob/Glob.cs]: I think this should go inside `GlobToRegexVisitor.Process` that way if we use the regex converter elsewhere we will get a consistent regex with the $. Could you make that change and I'll merge the PR?

Comment (dazinator) [inline src/Glob/Glob.cs]: Ok sure

**Comment by dazinator** [comment]
Done!

**Comment by dazinator** [comment]
Actually thinking about this.. it might not be the best solution.. what if the pattern ends with a * - then I guess you don't want to add a $ to the regex?

**Comment by dazinator** [comment]
I'll add a test case and find out

**Comment by dazinator** [comment]
Ok that doesn't appear to be a problem, but have added another test case for that anyway.

**Pull request merged**